src/dbally/context/context.py

```diff
@@ -0,0 +1,64 @@
+import ast
+from abc import ABC
+from typing import Iterable
+
+from typing_extensions import Self, TypeAlias
+
+from dbally.context.exceptions import ContextNotAvailableError
+
+CustomContext: TypeAlias = "BaseCallerContext"
+
+
+class BaseCallerContext(ABC):
+    """
+    An interface for contexts that are used to pass additional knowledge about
+    the caller environment to the filters. LLM will always return `Context()`
+    when the context is required and this call will be later substituted by an instance of
+    a class implementing this interface, selected based on the filter method signature (type hints).
+    """
+
+    _alias: str = "Context"
```

Comment:
nit: instead of providing a default value, I would require inheriting classes to define their own (throw exception otherwise or even better use an abstract method definition), to ensure that they don't use the default value by accident. Especially since it might be easy for people to assume that the name of the class itself is used as an alias automatically.